cmake/scripts/ROOTConfig.cmake.in

```diff
@@ -57,8 +57,11 @@ get_filename_component(_thisdir "${CMAKE_CURRENT_LIST_FILE}" PATH)
 # Configure the path to the ROOT headers, using a relative path if possible.
 # This is only known at CMake time, so we expand a CMake supplied variable.
 #
+@_ROOT_INCLUDE_DIR_SETUP@
```

Comment:
We want less of these build system variables to depend on. This makes ROOT relocate much harder. We have taken the route to reduce those and have ROOT compute its location dynamically.

Why do you need to export them here? Could you explain their intended use?

Reply:
I'm not sure what you mean. These macros are expanded at configure time and ROOT does not depend on them, because they don't exists at runtime. They split the path calculation that CMake (not ROOT) already does in two steps using the installed `ROOTConfig.cmake`. Nothing more is added and intermediate vars just serve as a second expansion to resolve the actual path at runtime and not what it currently returns `${_thisdir}/../../../usr/lib64/root`. What it currently does is broken on any system that does not install the whole of ROOT in one directory (the `$SYSROOT` way). Relevant issue [here](https://bugs.archlinux.org/task/64613) and [here](https://root-forum.cern.ch/t/root-6-18-not-searching-under-usr-local-include-for-header/37471). 

The reason it was done this was was exactly to do what ROOT already does, but transparently - meaning no new variables in config and ROOT or dependents of ROOT are none-the-wiser about anything changing.

However, I could probably make it do the same with a single `*_SETUP` macro.